.github/workflows/unittest.yaml

```diff
@@ -32,5 +35,135 @@ jobs:
+        uses: actions/download-artifact@v4
+        with:
+          name: artifact-conda-package
+          path: /tmp/local-channel/linux-64
+
+      - name: Install the package
+        run: |
+          micromamba install --yes -c conda-forge conda-build conda-index
+          python -m conda_index /tmp/local-channel
+          micromamba install --yes -c /tmp/local-channel -c mantid-ornl -c conda-forge ${{ env.PKG_NAME }}
+
+      - name: Verify the installation
+        run: |
+          conda_version=$(micromamba list "${{ env.PKG_NAME }}" | awk -v pkg="${{ env.PKG_NAME }}" '$1 == pkg { print $2 }')
+          echo "Conda version: $conda_version"
+          wheel_version=$(python -c "import ${{ env.PKG_NAME }}; print(${{ env.PKG_NAME }}.__version__)")
+          echo "Wheel version: $wheel_version"
+          if [ "$conda_version" != "$wheel_version" ]; then
+            echo "Version mismatch!"
+            exit 1
+          else
+            echo "Versions match."
+          fi
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: [tests, build, conda-verify]
+    if: startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup Pixi
+        uses: prefix-dev/setup-pixi@v0.8.10
+        with:
+          pixi-version: v0.48.2
+          manifest-path: pyproject.toml
+
+      - name: Download conda package artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: artifact-conda-package
+
+      - name: Upload package to anaconda
+        env:
+          ANACONDA_API_TOKEN: ${{ secrets.ANACONDA_TOKEN }}
+          IS_RC: ${{ contains(github.ref, 'rc') }}
+        run: |
+          # label is main or rc depending on the tag-name
+          CONDA_LABEL="main"
+          if [ "${IS_RC}" = "true" ]; then CONDA_LABEL="rc"; fi
+          echo pushing ${{ github.ref }} with label $CONDA_LABEL
+          pixi run anaconda upload --label $CONDA_LABEL --user neutrons ${{ env.PKG_NAME }}-*.conda
+
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    needs: tests
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup pixi
+        uses: prefix-dev/setup-pixi@v0.8.10
+        with:
+          pixi-version: v0.48.2
+          manifest-path: pyproject.toml
+
+      - name: Build pypi package
+        run: |
+          # build the package
+          VERSION=$(pixi run -- versioningit .)
+          pixi run pypi-build
+      # publish your distributions here (need to setup on PyPI first)
+      - name: Publish package distributions to PyPI
+        if: startsWith(github.ref, 'refs/tags/v')
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
Same, packaging publishing should be in its own workflow, it is easier to maintain.
Imagine we put all source code of mantid in a single header file, it will compile, but it would be a nightmare to maintain and modify.